nixos/modules/services/x11/colord.nix

```diff
@@ -26,11 +26,16 @@ in {
 
     systemd.packages = [ pkgs.colord ];
 
+    systemd.services.colord = {
+      serviceConfig = {
+        StateDirectory = "colord";
```

Comment:
The directory is already created by `tmpfiles.d`. Maybe we can just remove the home?

Reply:
I tried the following change on a system without `/var/lib/colord` and it still fails the startup: `CdMain: failed to load mapping database: Can't open database: unable to open database file`

```diff
--- a/nixos/modules/services/x11/colord.nix
+++ b/nixos/modules/services/x11/colord.nix
@@ -30,7 +30,6 @@ in {

     users.users.colord = {
       isSystemUser = true;
-      home = "/var/lib/colord";
       group = "colord";
     };

```

My guess is that `StateDirectory` had to be explicitly declared for a service to get write access to it if thinks like `ProtectHome` and `ProtectSystem` are on.

Reply:
I think that would make sense for `ProtectHome` to block access to the declared home directory. But if this does not work without it then it cannot work anywhere else either.

Reply:
Looks like they are actually doing this https://github.com/hughsie/colord/commit/08a32b2379fb5582f4312e59bf51a2823df56276

Reply:
Oh, and there are some other fixes https://github.com/hughsie/colord/commit/9283abd9c00468edb94d2a06d6fa3681cae2700d

Reply:
Proposed a rollback instead: https://github.com/NixOS/nixpkgs/pull/288378